plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,824 @@
+        if not (os.path.exists(path) or os.path.isdir(path)):
+            module.exit_json(msg=INVALID_DIRECTORY_MSG.format(path=path), failed=True)
+        file_path = module.params.get('share_parameters').get('share_name') + "/" + module.params.get('share_parameters').get('file_name')
+        file_exits = os.path.exists(file_path)
+        if file_exits:
+            with open(file_path, "rb") as cert:
+                cert_content = cert.read()
+                read_file = base64.encodebytes(cert_content).decode('ascii')
+        else:
+            module.exit_json(msg=NO_FILE_MSG, failed=True)
+        payload["LicenseFile"] = read_file
+        payload["FQDD"] = resource_id
+        payload["ImportOptions"] = "Force"
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __import_license_http(self, module, import_license_url, resource_id):
+        payload = {}
+        payload["LicenseName"] = module.params.get('share_parameters').get('file_name')
+        payload["FQDD"] = resource_id
+        payload["ImportOptions"] = "Force"
+        proxy_details = get_proxy_details(module)
+        payload.update(proxy_details)
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __import_license_cifs(self, module, import_license_url, resource_id):
+        payload = {}
+        payload["ShareType"] = "CIFS"
+        payload["LicenseName"] = module.params.get('share_parameters').get('file_name')
+        payload["FQDD"] = resource_id
+        payload["ImportOptions"] = "Force"
+        if module.params.get('share_parameters').get('workgroup'):
+            payload["Workgroup"] = module.params.get('share_parameters').get('workgroup')
+        share_details = get_share_details(module)
+        payload.update(share_details)
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __import_license_nfs(self, module, import_license_url, resource_id):
+        payload = {}
+        payload["ShareType"] = "NFS"
+        payload["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+        payload["ShareName"] = module.params.get('share_parameters').get('share_name')
+        payload["LicenseName"] = module.params.get('share_parameters').get('file_name')
+        payload["FQDD"] = resource_id
+        payload["ImportOptions"] = "Force"
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __check_file_extension(self, module):
+        share_type = module.params.get('share_parameters').get('share_type')
+        file_name = module.params.get('share_parameters').get('file_name')
+        valid_extensions = {".txt", ".xml"} if share_type == "local" else {".xml"}
```

Comment:
Any reason why you have extension in set and then converting them to tuples? Avoid unnecessary type casting of data

Reply:
Done